apps/backend/src/reconciliation/cash-reconciliation.service.ts

```diff
@@ -19,60 +20,62 @@ export class CashReconciliationService {
     @Inject(TransactionService) private transactionService: TransactionService
   ) {}
 
-  public async setMatched(
-    deposit: Partial<CashDepositEntity>,
-    payment: Partial<PaymentEntity>
-  ) {
-    if (!payment || !deposit) return;
-    return {
-      deposit: await this.cashDepositService.reconcileCash(deposit, payment),
-      payment: payment?.id?.split(',').map(async (itm) => {
-        return await this.transactionService.reconcileCash(deposit, {
-          ...payment,
-          id: itm
-        });
-      })
-    };
-  }
+  private allOrNoneMatchForPayments(
+    deposits: CashDepositEntity[],
+    payments: PaymentEntity[]
+  ): CashPaymentsCashDepositPair | void {
+    const sumOfPayments = payments.reduce(
+      (sum, payment) => sum + payment.amount,
+      0
+    );
 
-  public async match(deposits: CashDepositEntity[], payments: PaymentEntity[]) {
-    const matched = deposits.map((itm: CashDepositEntity) => ({
```

Comment:
There are several issues here: 

- typecasting in logic rather than class getters. 

- having partial entities restrict the amount of matching we can do here, this is still ok since we do only amts but will constrain us if the function has scope beyond it. 

- having DB queries in a loop same as that of the business logic. Although this is not a huge performance penalty now, it make it harder to process the logic. It also makes logging, auditing etc harder. 

- too many nested promises make debugging and control flow harder.